### ISSUE TITLE
Intro cinematic gets corrupted 

### ISSUE THREAD

**Issue by emptyVoid** [OpenApoc/OpenApoc#826]
Intro cinematic gets corrupted when UFO appears.

![image](https://user-images.githubusercontent.com/13639229/79841249-d2fcda80-83bf-11ea-8c1f-4691657a1f71.png)

These console messages appear during playback:
```
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 2 entries from 140 to 141
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 3 entries from 127 to 128
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 4 entries from 124 to 125
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 2 entries from 138 to 139
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 2 entries from 138 to 139
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 3 entries from 148 to 150
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 2 entries from 127 to 128
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 2 entries from 126 to 127
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 3 entries from 124 to 126
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 3 entries from 137 to 138
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 3 entries from 138 to 139
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 3 entries from 127 to 128
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 4 entries from 124 to 125
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 2 entries from 128 to 129
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 6 entries from 126 to 129
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 9 entries from 126 to 127
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 3 entries from 130 to 132
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 9 entries from 219 to 226
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 52 entries from 173 to 174
libsmacker::palette_render(s,p,size) - ERROR: overflow, 0x40 attempt to copy 6 entries from 24 to 25
```

[log.txt](https://github.com/OpenApoc/OpenApoc/files/4508568/log.txt)

**Comment by FilmBoy84** [comment]
Interesting, i cannot replicate this

Can you please send me your OpenApoc config file and let me know the source of the cd.iso you are using please?

Also GFX Card model and driver version

**Comment by JonnyH** [comment]
IIRC some palette overflows are normal - at least I see a couple at the start.

What version of apoc are you using for the data files? The steam one? gog.com? Actual CDs? Something Else (if the answer to that is "piracy" then the only answer is to get a known-good copy, many pirated copies tried to 'trim' things they thought were unneeded, so aren't actually a good copy of the game. Also it's illegal)

But it might be useful to check the md5sum of the video filesas read from the cd or something, in case they got corrupted. For reference (from my steam iso):

c5fedf277517fc856d49b9d8b0a178ea  intro1.smk

**Comment by Atrosha** [comment]
The same *** I thought this normal.
On nVIDIA GTX 1060 and AMD RX 5700.

And there is the screencast: https://youtu.be/abWZpoKmSc0

cd.iso from Steam.

**Comment by emptyVoid** [comment]
[OpenApoc_settings.zip](https://github.com/OpenApoc/OpenApoc/files/4512077/OpenApoc_settings.zip)

I'm using `cd.iso` from Steam (downloaded it yesterday). Hash is the same:
```
PS C:\Users\emptyVoid> Get-FileHash "E:\XCOM3\SMK\INTRO1.SMK" -Algorithm MD5

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
MD5             C5FEDF277517FC856D49B9D8B0A178EA                                       E:\XCOM3\SMK\INTRO1.SMK
```

GeForce GTX 1080 Ti / Driver 445.87

**Comment by JonnyH** [comment]
Its entirely possible it's got broken at some point - I can't even remember the last time I didn't instantly skip the intro

**Comment by FilmBoy84** [comment]
Ah, replicated with Steam.iso

The log reports the overflow when the screen is black, just before "Punch it in" but the overflow only becomes visually apparent to the end user at the end of the cached material (which is approximately when the UFO appears)

![image](https://user-images.githubusercontent.com/33913088/79903863-b71f2600-840b-11ea-92c2-58e3b6dda096.png)

**Comment by JonnyH** [comment]
There's an upstream fix for palette issues in libsmacker - PR #828 may help?

**Comment by JonnyH** [comment]
Even with that fix I seem to get messed up palettes in some scenes - I guess I'll have to debug libsmacker - as as far as I can see the palettes we are using are the ones that's returning to us

**Comment by JonnyH** [comment]
Yes, that upstream fix is incomplete - it still blocks palette copies that happen to to overlap their source and destination regions - despite the upstream fix now making that well defined.

I've made that fix to the libsmacker clone used in the dependencies/ directory, and updated the PR.

Hopefully that should fix everything? It certainly looks better to my eyes

**Comment by JonnyH** [comment]
Fixed in #828 

Please reopen if issues remain
